### PR TITLE
httpbakery: allow service to set cookie location for macaroons

### PR DESCRIPTION
**Pull request by rogpeppe**
This allows a service to return a discharge-required error
that will be presented for more than just the original
request.

### CODE REVIEW

File: httpbakery/client.go [hunk 1]
```diff
@@ -184,9 +184,18 @@ func (ctxt *clientContext) do1(req *http.Request, getBody BodyGetter) (*http.Res
...
 	macaroons, err := bakery.DischargeAll(mac, ctxt.obtainThirdPartyDischarge)
 	if err != nil {
-		return nil, err
+		return nil, errgo.Mask(err, errgo.Any)
+	}
+	cookieURL := req.URL
+	if path := resp.Info.MacaroonPath; path != "" {
+		relURL, err := parseURLPath(path)
+		if err != nil {
+			logger.Errorf("ignoring invalid path in discharge-required response: %v", err)
```
Comment (mhilton) [inline httpbakery/client.go]: Warning?

**Comment by frankban** [comment]
:+1: thank you!

**Comment by mhilton** [comment]
:+1: LGTM

File: httpbakery/client.go (hunk 1 above)
Comment (rogpeppe) [inline httpbakery/client.go]: done

**Pull request merged**